tinygrad/helpers.py

```diff
@@ -305,7 +305,7 @@ def to_mv(ptr, sz) -> memoryview: return memoryview(ctypes.cast(ptr, ctypes.POIN
 def mv_address(mv:memoryview): return ctypes.addressof(ctypes.c_char.from_buffer(mv))
 def to_char_p_p(options: List[bytes], to_type=ctypes.c_char): return (ctypes.POINTER(to_type) * len(options))(*[ctypes.cast(ctypes.create_string_buffer(o), ctypes.POINTER(to_type)) for o in options])  # noqa: E501
 @functools.lru_cache(maxsize=None)
-def init_c_struct_t(fields: Tuple[Tuple[str, ctypes._SimpleCData], ...]):
+def init_c_struct_t(fields: Tuple[Union[Tuple[str, Type[ctypes._SimpleCData]], Tuple[str, Type[ctypes._SimpleCData], int]], ...]):
```

Comment:
Why do you need `Type[`?

Reply:
Because what ends up being passed into this function aren't instances of `_SimpleCData` itself, it's instances of it's subclasses, thus typeguard crashes test_ops.